packages/libs/error-stack/src/ext/iter.rs

```diff
@@ -0,0 +1,297 @@
+use core::fmt::{Debug, Display};
+
+use crate::{Context, Report, ResultExt};
+
+/// This trait provides extension methods for iterators which makes it possible to turn them into
+/// `Report`s.
+///
+/// This method works only on iterators of items which implement [`crate::ResultExt`].
+///
+/// Because this trait attaches the provided data to _every_ item in the iterator, the
+/// provided data must implement [`Clone`].
+pub trait IteratorExt: Iterator + Sized {
+    /// Adds a new attachment to each [`Report`] in the iterator.
```

Comment:
In the `future` crate we have more extensive documentation for those methods. Can we adopt those here as well?